sentry/src/main/java/io/sentry/SentryAutoDateProvider.java

```diff
@@ -0,0 +1,35 @@
+package io.sentry;
+
+import io.sentry.util.Platform;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class SentryAutoDateProvider implements SentryDateProvider {
+
+  private final @NotNull SentryDateProvider dateProvider;
+
+  public SentryAutoDateProvider() {
+    if (checkInstantAvailabilityAndPrecision()) {
```

Comment:
`h`: is it just me or the if-check should be inverted here? 😂 if Instant is available with nanos precision we should use `SentryInstantDateProvider`?

Reply:
Oopsie 🤣

Reply:
btw, I think you can also test it, because you could set `System.setProperty("java.specification.version", version)` in tests I believe

Reply:
Added a new test